src/expression.js

```diff
@@ -206,7 +206,7 @@ pp.parseExprSubscripts = function(refShorthandDefaultPos) {
   let startPos = this.start, startLoc = this.startLoc
   let expr = this.parseExprAtom(refShorthandDefaultPos)
   if (refShorthandDefaultPos && refShorthandDefaultPos.start) return expr
-  return this.parseSubscripts(expr, startPos, startLoc)
+  return this.parseSubscripts(expr, startPos, startLoc, expr.type === "ArrowFunctionExpression" && this.parenthesesCount === 0)
```

Comment:
I mean in that function, is there any case (not just calls) where we do want subscripts after `() => 42` to be parsed? I assume they all are already handled as part of arrow function's body? Otherwise we currently still parse `() => {}.x` as valid syntax, not sure if that's correct...